blaze/expr/collections.py

```diff
@@ -708,8 +710,45 @@ def isin(expr, keys):
     return IsIn(expr, frozenset(keys))
 
 
+class Shift(ElemWise):
+    """ Shift a column backward or forward by N elements
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression to shift. This expression's dshape should be columnar
+    n : int
+        The number of elements to shift by. If n < 0 then shift backward,
+        if n == 0 do nothing, else shift forward.
+    """
+    __slots__ = '_hash', '_child', 'n'
+
+    @property
+    def schema(self):
+        measure = self._child.schema.measure
+
+        # if we are not shifting or we are already an Option type then return
+        # the child's schema
+        if not self.n or isinstance(measure, Option):
+            return measure
+        else:
+            return Option(measure)
+
+    def __str__(self):
+        return '%s.%s(n=%d)' % (
+            self._child, type(self).__name__.lower(), self.n
+        )
+
+
+@copydoc(Shift)
+def shift(expr, n):
+    if not isinstance(n, (numbers.Integral, np.integer)):
```

Comment:
is there a reason we cannot allow expressions for n. like, `a.shift(b.count())`?